packages/fcl/src/interaction-template-utils/derive-cadence-by-network.js

```diff
@@ -0,0 +1,69 @@
+import {invariant} from "@onflow/sdk"
+import {normalizeInteractionTemplate} from "./normalize/interaction-template"
+
+export function deriveCadenceByNetwork({network, template}) {
+  invariant(
+    network != undefined,
+    "deriveCadenceByNetwork({ network }) -- network must be defined"
+  )
+  invariant(
+    typeof network === "string",
+    "deriveCadenceByNetwork({ network }) -- network must be a string"
+  )
+
+  invariant(
+    template != undefined,
+    "generateDependencyPin({ template }) -- template must be defined"
+  )
+  invariant(
+    typeof template === "object",
+    "generateDependencyPin({ template }) -- template must be an object"
+  )
+  invariant(
+    template.f_type === "InteractionTemplate",
+    "generateDependencyPin({ template }) -- template must be an InteractionTemplate"
+  )
+
+  template = normalizeInteractionTemplate(template)
+
+  switch (template.f_version) {
```

Comment:
Is this all the same as `deriveDependenciesByNetwork`?

Reply:
It's close, but instead of returning dependencies, `deriveCadenceByNetwork` returns cadence